extensions/build-gradle/main.ts

```diff
@@ -0,0 +1,366 @@
+
+    if(subproject) {
+        cmd = subproject + ":dependencies";
+    }
+
+    let gradleResp = await invokeGradle(["-q", cmd]);
+
+    if (!gradleResp) {
+        return;
+    }
+
+    // Convert the Uint8 array to ascii
+    const ret = new TextDecoder().decode(gradleResp);
+
+    if(ret.indexOf("BUILD FAILED") > 0) {
+        return;
+    }
+
+    return ret;
+}
+
+async function invokeGradle(cmd) {
+    //Try our directory
+    try {
+        return await Deno.run({
+            cmd: ["./gradlew"].concat(cmd),
+            stdout: "piped"
+        }).output();
+    } catch(e) {
+        //doNothing()
+    }
+
+    //Try single parent directory
+    try {
+        return await Deno.run({
+            cmd: ["../gradlew"].concat(cmd),
+            stdout: "piped"
+        }).output();
+    } catch(e) {
+        //doNothing()
+    }
+
+    //Lastly try using gradle directly
+    try {
+        return await Deno.run({
+            cmd: ["gradle"].concat(cmd),
+            stdout: "piped"
+        }).output();
+    } catch(e) {
+        //doNothing()
+    }
+
+    logError("ERROR: It doesn't look like you have `gradle` installed or " +
+             "gradle wrapper in the current or parent directories");
+}
+
+/**
+ *  Parse a `build.gradle` file and returned the identified dependencies. 
+ */
+async function getBuildGradleDeps(subproject: string) {
+    const gradleDeps = await generateGradleDeps(subproject);
+
+    if(!gradleDeps) {
+        logError("ERROR: Failed to parse dependencies. Check your " +
+                      "`build.gradle` file."); 
+        return;
+    }
+
+    // Parse the dependencies from this file.
+    let foundDeps = parseGradleFile(gradleDeps); 
+
+    return foundDeps;
+}
+
+/**
+ *  Submit the provided dependencies to Phylum for analysis.
+ */
+async function submit(pkgs: object[], project: string, group: string) {
+    if(!pkgs.length) {
+        return;
+    }
+
+    logSuccess("Submitting to Phylum for analysis...");
+        
+    if(!project) {
+        logSuccess("Please specify a project with --project <projectName>");
+    }
+
+    if(group && !project) {
+        logError("ERROR: You cannot specify a group without a project.");
+        return;
+    }
+
+    if(project) {
+        logSuccess(`\t --> Project: ${project}`)
+    }
+
+    if(group) {
+        logSuccess(`\t --> Group: ${group}`)
+    }
+
+    if(!group && !project) {
+        logError("ERROR: You must specify a project (and optionally a group).");
+        return;
+    }
+
+    if(pkgs.length) {
+        const jobId = await PhylumApi.analyze("maven", pkgs, project, group);
+        logSuccess(`Job submitted for analysis with job ID: ${jobId}`);
+    } else {
+        logSuccess("No packages to submit");
+    }
+}
+
+/**
+ *  The extension API does not currently provide facilities for creating a project.
+ *  Shim out to the Phylum CLI to handle this (inception!).
+ */
+async function attemptCreateProject(projectName: string, group?: string) {
```

Comment:
This is too hacky for an official extension. Our extensions should avoid calling back in to the `phylum` binary like this.

For the initial version, at least, I think it would be simplest if this extension did not automatically create projects.